testng-core/src/main/java/org/testng/TestClass.java

```diff
@@ -216,21 +235,29 @@ private void initMethods() {
               false,
               eachInstance);
       m_beforeTestMethods =
-          ConfigurationMethod.createTestMethodConfigurationMethods(
-              objectFactory,
-              testMethodFinder.getBeforeTestMethods(m_testClass),
-              annotationFinder,
-              true,
-              xmlTest,
-              eachInstance);
+          Stream.of(
```

Comment:
That's a lot of logic just to deal with arrays. Could you try to replace some array by appropriate collection and keep arrays on interface method implementations only?

Reply:
Fixed